### ISSUE TITLE
DISCUSS: scipy version number after 1.9

### ISSUE THREAD

**Issue by h-vetinari** [scipy/scipy#15528]
Scipy has a tried-and-tested release mode that closely mimics numpy: no semver, but rolling additions & deprecations & eventual removals.

With `main` now being at 1.9, certain deprecations are running into the problem of how to announce the point-in-time when a given feature will be removed. Since the possibility of Scipy 2.0 turned up in a [discussion](https://github.com/scipy/scipy/pull/13490#discussion_r790426384) recently, coming up with good wording for this is tricky.
1. "X will be removed in the second release after SciPy 1.9.0"
   * currently used by #13490 and #15512 
   * IMO misleading/wrong because "second release after SciPy 1.9.0" == 1.9.2
2. variations on that theme are of course possible (additions emphasised for clarity)
   * "X will be removed in the second _**minor**_ release after SciPy 1.9.0" (not a good fit because scipy doesn't use semver)
   * "X will be removed in the second _**feature**_ release after SciPy 1.9.0" ([suggestion](https://github.com/scipy/scipy/pull/15512#discussion_r800002877) by @mdhaber)
3. directly say "X will be removed in SciPy 1.11.0"
   * if a decision is taken towards releasing 2.0, this could be easily replaced

However, all those issues effectively boil down to an uncertainty about what to call the version after 1.9, hence this issue.

I had assumed (perhaps naïvely) that there's no reason to break the current scheme, not least because version numbers are not decimals and _easily_ deal with two-digit minor versions, but perhaps more importantly, because numpy and cpython did so as well. My [comment](https://github.com/scipy/scipy/pull/15512#discussion_r800019349) about this in the second thread was:
> I'm thinking a putative 2.0 should come with some degree of "large change compared to 1.x", even if it isn't breaking anything**.
>
> ** as an example in terms of the kind of scope for a major bump that I have in mind, a current(ly-talked-about) candidate for cpython 4.0 is if/once the GIL is removed. On the numpy side, although there is an ancient, non-empty [milestone](https://github.com/numpy/numpy/milestone/9), I'm not aware of current plans for 2.0 (though I did once [wager](https://github.com/numpy/numpy/issues/12525#issuecomment-452463450) that it'll come about eventually 😉) - comprehensively fixing integer promotion issues _might_ be a candidate to go to numpy 2.0.

In other words, I would expect the 1.x series to continue indefinitely unless & until there's a large enough change that people feel deserves special attention by way of a major version bump.

Howevers, others might disagree; e.g. @ev-br [said](https://github.com/scipy/scipy/pull/15512#discussion_r800034055):
> My personal opinion is that 2.0 is no special and I'd have a slight preference for 1.9->2.0. That said, it's not quite my hill so I'd go with a majority on this one :-)

**Comment by ilayn** [comment]
NumPy 2.0 is a bit what SciPy 1.0 was. A mostly psychological milestone that signals somewhat maturity or a well-earned achievement trophy. SciPy 2.0 is not the same, as @ev-br put it, as 1.0,  a rather follow-up. Another extreme numbering package was pandas, same with Cython from 0.29 to straight 3.0. 

To be honest I don't get what fuss is about version numbers in Python and seems like some people just satisfying their OCD. The idea is noble but the practice has turned out to be not so useful to put it mildly. I would suggest continuing 10, 11, 12 since nobody cared for other very important packages. But honestly I couldn't care less.

**Comment by Kai-Striega** [comment]
I'm +1 on continuing with the current scheme (1.10, 1.11, ... etc) and reserving 2.0 for some significant milestone.

There are actually some references in the code base to a SciPy 2.0 release e.g. in `scipy/stats/stats.py` there's the line:
> This file is not meant for public use and will be removed in SciPy v2.0.0.

**Comment by mdhaber** [comment]
Ditto on continuing the current scheme until support for designating a release as 2.0 has been bubbling for a little while.

**Comment by ev-br** [comment]
(Let me expand my comment over from the other issue.) 

Note that both numpy 2.0 and cpython 4.0, which were mentioned before, are essentially shorthands for "not in a foreseeable furure".

IMO a two-digit scheme has its advantages, and we can perfectly well go from 1.9 to 2.0. There is some amount of back compat breakage in each release, which we are very measured about, and 2.0 does not need to be any special. 

If we do decide to assign some special meaning to the 2.0 milestone, we need to explicitly formulate what makes it special and list specific items which make scipy 2.0-ready.

**Comment by tupui** [comment]
I am with @ev-br here. I am for not flagging a special breaking release but do continuous improvement as we currently do. Going above 2 would highlight this statement of stability from our side. And also say that user should move with us. We certainly don't want users to be stuck with a certain 1.1x with and not go with the hypothetical 2.x which would break things. This would just create fragmentation as always.

**Comment by perimosocordiae** [comment]
If we're straw-polling, I'm +1 for 1.10 after 1.9.

I've personally been thinking we might want to remove the spmatrix API with scipy 2.0, among other big-ticket deprecations.

**Comment by WarrenWeckesser** [comment]
I also think 1.9 should be followed by 1.10.

I'll add a reminder that for 2.0.0, we have the commitment to remove all the numpy names that are currently imported into the scipy namespace; see https://github.com/scipy/scipy/pull/10290 and the code in `scipy/__init__.py`.

**Comment by tylerjereddy** [comment]
Sounds like we're pretty happy with `1.10`, `1.11`, etc. for now. Me too.

**Comment by rgommers** [comment]
I didn't see this till now; +1 for continuing as normal with `1.10`. Beyond some users using incorrect string comparisons causing an issue with `'1.10' > '1.9'` being false (which is a minor thing), I see no reason to not do `1.10.0`.

**Comment by h-vetinari** [comment]
Seeing that there seems to be relatively broad agreement on continuing with `1.x` for the time being, I think this discussion has run its course?

As an immediate outcome, I'd like to propose creating milestones for 1.10 & 1.11 already - in the past we've missed promised deprecations, and having such future milestones would allow us to file and milestone issues for a given action-to-be-taken already now (i.e. something gets deprecated for 1.9 with promised removal in 1.11, and we already file an issue against 1.11 as a definite reminder to ourselves to remove said functionality).

**Comment by tylerjereddy** [comment]
> I'd like to propose creating milestones for 1.10 & 1.11 already

Ok, I've gone ahead and created those milestones.

I have to admit I tend to focus what volunteer time I have mostly on PRs that are open with a milestone rather than issues, but hopefully others will occasionally scan the milestoned deprecation-removal issues and then open a PR to do the final removal within a few weeks-months of the next release--then I'd be unlikely to miss the PR.

I suppose those PRs would typically be straightforward to merge anyway.

**Comment by h-vetinari** [comment]
Cool thanks! I think this issue can thus be closed.

**Comment by h-vetinari** [comment]
While looking for deprecation warnings, I just noticed that https://github.com/scipy/scipy/pull/14919 promises the removal of several public modules by 2.0.0. I think that's compatible with the discussion we had above, but just wanted to call attention to it. I'll add it to the 2.0 wiki.

**Comment by rgommers** [comment]
Yes, that just meant "not in 2 releases, we're leaving it in for longer"
